### DISCUSSION TITLE
Passing data from back from executable to orchestrator

### DISCUSSION THREAD

**Discussion by rosieks** [microsoft/aspire#5360] [Q&A] [answered]
I'm trying to automate creation of Keycloak's realm. I have already separate project similar to one for running EF migrations.
And now I stuck on the case where I create client to manage KeyCloak relam from my application level and I have to pass client credentials back as parameter for application as an environment variable. Any idea how to achieve that?

**Comment by davidfowl** [comment]
When you say "from my application level", do you mean the apphost? Can you show the code you have?

**Reply by rosieks** [reply]
So my code looks like below:

```csharp
var keycloakClientSecret = builder.AddParameter("keycloakClientSecret");

// Keycloak
var keycloak = builder.AddKeycloakContainer("keycloak");
var realm = keycloak.AddRealm("cbt-realm", realm: "cbt");

// Setup keycloak
var keycloakSetup = builder.AddProject<Projects.KeycloakSetup>("keycloak-setup")
    .WithReference(keycloak).WithReference(realm);

// My app
var api = builder.AddProject<Projects.MyApi>("api")
   .WithReference(keycloak).WithReference(realm)
   .WithEnvironment("Keycloak__ClientId", "provisioning_client")
   .WithEnvironment("Keycloak__ClientSecret", keycloakClientSecret)
```

1. I have Keycloak container
2. *keycloak-setup* configure the whole realm in Keycloak. It also create new client called *provisioning-client*.
3. *api* is my application that manages Keycloak realm. 

The issue that I have is that value of *keycloakClientSecret* is created inside of *keycloak-setup* and I would like to somehow pass it as parameter. Right now I just log it and then manually copy-paste it to *appsettings.json* (probably I should move it to *secrets.json*)

**Reply by davidfowl** [reply, marked as answer]
Can you instead generate the keycloakClientSecret in the apphost and pass it to both the "keycloak-setup" and "api"?

**Reply by rosieks** [reply]
Actually I see it's possible in Keycloak to provide own client secret. Thanks

Anyway I think it's nice to have such option in the future just in case someone wants to achieve something similar.

**Reply by davidfowl** [reply]
That's a shared state problem. You need both projects to look at the same data for the code. If project A needs data from project B, then project B needs to read it from the place that project A writes it to.
